cmd/mdatagen/metric-metadata.yaml

```diff
@@ -25,6 +25,8 @@ metrics:
     unit:
     # Required: metric type with its settings.
     <sum|gauge|histogram>:
+      # Required for sum and gauge metrics: type of number data point values.
+      number_type: # int | double
```

Comment:
In the collector pdata we call this "value_type" because may be possible that we may support "enum" or "string" etc. https://github.com/open-telemetry/opentelemetry-collector/blob/main/model/pdata/metrics.go#L309 I think for consistency we should call this value_type.

Reply:
Makes sense. Updated